File: src/main/java/moe/giga/discord/commands/Command.java
package moe.giga.discord.commands;

import moe.giga.discord.contexts.MessageContext;
import net.dv8tion.jda.core.JDABuilder;

public abstract class Command {
    public abstract void onCommand(MessageContext MC, String[] args);

    public void init(JDABuilder builder) {

    }
}

File: src/main/kotlin/moe/giga/discord/commands/Command.kt
package moe.giga.discord.commands

import moe.giga.discord.contexts.MessageContext
import moe.giga.discord.util.AccessLevel
import net.dv8tion.jda.core.JDABuilder

interface Command {
    fun execute(MC: MessageContext, args: List<String>)

    fun init(builder: JDABuilder) {

    }

    val name: String
        get() = ""
    val description: String
        get() = ""
    val usage: String
        get() = name
    val alias: String?
        get() = null
    val hidden: Boolean
        get() = false
    val level: AccessLevel
        get() = AccessLevel.USER
}
